nixos/modules/services/misc/recyclarr.nix

```diff
@@ -0,0 +1,166 @@
+{
+  config,
+  lib,
+  pkgs,
+  utils,
+  ...
+}:
+
+let
+  cfg = config.services.recyclarr;
+  format = pkgs.formats.yaml { };
+  configPath = "/var/lib/recyclarr/config.json";
+in
+{
+  options.services.recyclarr = {
+    enable = lib.mkEnableOption "recyclarr service";
+
+    package = lib.mkPackageOption pkgs "recyclarr" { };
+
+    configuration = lib.mkOption {
+      type = format.type;
+      default = { };
+      example = lib.literalExpression ''
+        {
+          sonarr = [{
+            instance_name = "main";
+            base_url = "http://localhost:8989";
+            api_key = {
+              _secret = "/run/credentials/recyclarr.service/sonarr-api_key";
+            };
+          }];
+          radarr = [{
+            instance_name = "main";
+            base_url = "http://localhost:7878";
+            api_key = {
+              _secret = "/run/credentials/recyclarr.service/radarr-api_key";
+            };
+          }];
+        }
+      '';
+      description = lib.mdDoc ''
+        Recyclarr YAML configuration as a Nix attribute set.
+
+        For detailed configuration options and examples, see the
+        [official configuration reference](https://recyclarr.dev/wiki/yaml/config-reference/).
+
+        The configuration is processed using [utils.genJqSecretsReplacementSnippet](https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/utils.nix#L232-L331) to handle secret substitution.
+
+        To avoid permission issues, secrets should be provided via systemd's credential mechanism:
+
+        ```nix
+        systemd.services.recyclarr.serviceConfig.LoadCredential = [
+          "radarr-api_key:''${config.sops.secrets.radarr-api_key.path}"
+        ];
```

Comment:
Oh that's a pretty nice way to handle this.

I recall @Stunkymonkey mentioned wanting to work towards a reusable and automated way to do this (`LoadCredential`) for any module needing secrets.

@Stunkymonkey this seems relevant to your interests, if you also want to review it.